pkg/ibgu/ibgu_test.go

```diff
@@ -432,3 +440,200 @@ func generateIbgu() *v1alpha1.ImageBasedGroupUpgrade {
+			assert.Equal(t, testIbgu.Namespace, builderResult.Object.Namespace)
+		}
+	}
+}
+
+func TestIbguDeleteAndWait(t *testing.T) {
+	testCases := []struct {
+		testibgu      *IbguBuilder
+		expectedError error
+	}{
+		{
+			testibgu:      buildValidIbguTestBuilder(buildTestClientWithDummyIbguObject()),
+			expectedError: nil,
+		},
+		{
+			testibgu:      buildValidIbguTestBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError: fmt.Errorf("ibgu cannot be deleted because it does not exist"),
+		},
+		{
+			testibgu:      buildInvalidibguTestBuilder(buildTestClientWithDummyIbguObject()),
+			expectedError: fmt.Errorf("ibgu 'nsname' cannot be empty"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		_, err := testCase.testibgu.DeleteAndWait(time.Second)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testibgu.Object)
+			assert.Nil(t, testCase.testibgu.Object)
+		}
+	}
+}
+
+func TestIbguWaitUntilDeleted(t *testing.T) {
+	testCases := []struct {
+		testibgu      *IbguBuilder
+		expectedError error
+	}{
+		{
+			testibgu:      buildValidIbguTestBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError: nil,
+		},
+		{
+			testibgu:      buildValidIbguTestBuilder(buildTestClientWithDummyIbguObject()),
+			expectedError: context.DeadlineExceeded,
+		},
+		{
+			testibgu:      buildInvalidibguTestBuilder(buildTestClientWithDummyIbguObject()),
+			expectedError: fmt.Errorf("ibgu 'nsname' cannot be empty"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		err := testCase.testibgu.WaitUntilDeleted(time.Second)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testibgu.Object)
+		}
+	}
+}
+
+func buildTestClientWithDummyIbguObject() *clients.Settings {
+	return clients.GetTestClients(clients.TestClientParams{
+		K8sMockObjects:  buildDummyibguObject(),
+		SchemeAttachers: testSchemes,
+	})
+}
+
+func buildDummyibguObject() []runtime.Object {
+	return append([]runtime.Object{}, buildDummyIbgu(defaultIbguName, defaultIbguNsName))
+}
+
+func buildDummyIbgu(name, namespace string) *v1alpha1.ImageBasedGroupUpgrade {
+	return &v1alpha1.ImageBasedGroupUpgrade{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.ImageBasedGroupUpgradeSpec{
+			ClusterLabelSelectors: make([]metav1.LabelSelector, 0),
+		},
+	}
+}
+
+// buildValidIbguTestBuilder returns a valid IbguBuilder for testing purposes.
+func buildValidIbguTestBuilder(apiClient *clients.Settings) *IbguBuilder {
+	return NewIbguBuilder(
+		apiClient,
+		defaultIbguName,
+		defaultIbguNsName,
+	)
+}
+
+// buildinInvalidibguTestBuilder returns an invalid ibguBuilder for testing purposes.
+func buildInvalidibguTestBuilder(apiClient *clients.Settings) *IbguBuilder {
+	return NewIbguBuilder(
+		apiClient,
+		defaultIbguName,
+		"")
+}
```

Comment:
it looks like there's already some generate* functions in this file that can do the same thing as these

Reply:
I removed the build* functions I copied from cpu and changed the tests I added to use the existing generate* functions